src/south/db/generic.py

```diff
@@ -639,7 +639,7 @@ def create_unique(self, table_name, columns):
         return name
 
     @invalidate_table_constraints
-    def delete_unique(self, table_name, columns):
+    def delete_unique(self, table_name, columns, if_exist=False):
```

Comment:
```suggestion
    def delete_unique(self, table_name, columns, required=False):
```

or maybe `optional=true` or `should_exist` etc.?

Reply:
Agreed, should_exist is better name